From 56655d08f70d7c27218c1210dcc3613bc318cac0 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Fri, 29 Nov 2024 01:19:50 +0100
Subject: [PATCH] [doged] Turn on legacyscriptrules by default on regtest (#92)

We want to have a uniform behavior across the different chains, else testing becomes really difficult.

diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index 0e38bee43a9a7ffe3c98ce709c3fa77336b13ae9..01efc121f50e00015a6a7c6f5c7ec7c6c02ab3e5 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -457,8 +457,8 @@ public:
         consensus.digishieldMinDiffHeight = 157500;
         // keep maturity same as Bitcoin for tests
         consensus.initialCoinbaseMaturity = REGTEST_COINBASE_MATURITY;
-        // legacy rules disabled for regtest so we don't refactor the universe
-        consensus.enforceLegacyScriptRules = false;
+        // legacy rules enabled for regtest
+        consensus.enforceLegacyScriptRules = true;
 
         diskMagic[0] = 0x94;
         diskMagic[1] = 0xb1;
diff --git a/test/functional/test_framework/util.py b/test/functional/test_framework/util.py
index e654e4137ad1d146ece9a93aa50c005c6bfdb284..7dbd86009ee7e7e33466b01df91f6aeb73571de8 100644
--- a/test/functional/test_framework/util.py
+++ b/test/functional/test_framework/util.py
@@ -578,6 +578,7 @@ def write_config(config_path, *, n, chain, extra_config="", disable_autoconnect=
         f.write("printtoconsole=0\n")
         f.write("natpmp=0\n")
         f.write("usecashaddr=1\n")
+        f.write("legacyscriptrules=0\n")
         # Increase peertimeout to avoid disconnects while using mocktime.
         # peertimeout is measured in mock time, so setting it large enough to
         # cover any duration in mock time is sufficient. It can be overridden
